wavelength-plugins/friends.js

```diff
@@ -0,0 +1,117 @@
+		if (fc === thisFC) {
+			let curFriends = Db.friend.get(user, []);
+			curFriends.push(target);
+			Db.friend.set(user, curFriends);
+			user.popup(`You have added ${target} as a friend.`);
+			return;
+		} else {
+			user.popup(`${fc} is not the right friendcode for ${target}.`);
+			return;
+		}
+	},
+	removeFriend: function (target, user) {
+		target = toId(target);
+		if (!Db.friend.get(user).includes(target)) return this.errorReply('You are not friends with this user.');
+		let userid = user.userid;
+		let curFriends = Db.friend.get(userid, []);
+		if (!curFriends.length) return user.popup("You don't have any friends to remove.");
+		let targetIndex = curFriends.findIndex(curFriend => toId(target) === curFriend);
+		if (targetIndex <= 0) return user.popup("You are not friends with this user.");
+		curFriends.splice(targetIndex, 1);
+		Db.friend.set(userid, curFriends);
+		user.popup(`You are no longer friends with "${target}".`);
+		return user.reply(`You are no longer friends with ${target}.`);
+	},
+};
+exports.commands = {
+	fc: {
+		make: function (target, room, user) {
+			let hasFC = Db.fc.get(user);
+			if (!hasFC) {
+				if (user.locked) this.errorReply('You can\'t obtain your friend code while locked.');
+				let fc = genFC();
+				Db.fc.set(user.userid, fc);
+				this.popupReply(`Your friend code is ${fc}. If you forget it, use /fc get .`);
+			} else {
+				this.popupReply('You already have a friend code.');
+			}
+		},
+		get: function (target, room, user) {
+			if (user.locked) this.errorReply('You can\'t give out your code while locked.');
+			let getfc = FC.getFC(user);
+			this.sendReplyBox(`Your friend code is: ${getfc}`);
+		},
+		help: function (target, room, user) {
+			this.sendReplyBox('<h2> Friend Code Commands: </h2><br />' + '<i> /fc make </i> - Makes your friend code. <br />' + '<i> /fc get </i> - Displays your friend code (in case you forget.)');
+		},
+	},
+	friends: 'friend',
+	friend: {
+		add: function (target, room, user) {
+			if (user.locked) this.errorReply('You can\'t make friends while locked.');
+			let [targetUser, targetCode] = target.split(',').map(p => toId(p));
+			if (!targetUser || !targetCode) return this.errorReply('/friend add [friend], [fc]');
+			if (!FC.getFC(targetUser)) return this.errorReply('Tell your friend to get a FC. /fc make');
+			if (targetUser < 1) return this.errorReply('Invalid user.');
+			if (targetUser > 19) return this.errorReply('Invalid user.');
+			Friend.addFriend(targetUser, targetCode, user);
+		},
+		remove: function (target, room, user) {
+			if (!target) return this.errorReply('/friend remove [friend]');
+			if (target < 1) return this.errorReply('Invalid user.');
+			if (target > 19) return this.errorReply('Invalid user.');
+			Friend.removeFriend(target, user);
+		},
+		removeall: function (target, room, user) {
+			if (Db.friend.get(user).length === 0) this.errorReply('You have no friends.');
+			Db.friend.set(Db.friend.get(user), []);
```

Comment:
Don't use `Db.friend.get(user)` as the key, or you're setting the value of their friends to `[]`.

Par exemple,
```js
Db.friend.set("mystifi", ["hoeenhero"]);
let mystFriends = Db.friend.get("mystifi"); // ["hoeenhero"]
Db.friend.set(mystFriends, []); // setting [] to an array?
```
Doesn't really make sense. Just do this:
```js
Db.friend.set(user, []);
```
After you make sure that `user` is the user's userid, of course.